packages/iov-ethereum/src/serialization.ts

```diff
@@ -108,6 +110,59 @@ export class Serialization {
           chainIdHex,
         );
       }
+    } else if (isSwapOfferTransaction(unsigned)) {
+      const chainIdHex = encodeQuantity(fromBcpChainId(unsigned.creator.chainId));
+      if (!unsigned.fee || !unsigned.fee.gasPrice) {
+        throw new Error("fee.gasPrice must be set");
+      }
+      const gasPriceHex = encodeQuantityString(unsigned.fee.gasPrice.quantity);
+      if (!unsigned.fee.gasLimit) {
+        throw new Error("fee.gasLimit must be set");
+      }
+      const gasLimitHex = encodeQuantityString(unsigned.fee.gasLimit.quantity);
+
+      if (!unsigned.swapId) {
+        throw new Error("No swap ID provided");
+      }
+
+      if (!unsigned.contractAddress || !isValidAddress(unsigned.contractAddress)) {
+        throw new Error("Invalid contract address");
+      }
+
+      if (!isValidAddress(unsigned.recipient)) {
+        throw new Error("Invalid recipient address");
+      }
+
+      if (unsigned.memo) {
+        throw new Error("Memo cannot be serialized in an atomic swap offer transaction");
+      }
+
+      if (!(unsigned.timeout as BlockHeightTimeout).height) {
```

Comment:
`!isBlockHeightTimeout(unsigned.timeout)` should make the cast below unnecessary